addons/portal/controllers/portal.py

```diff
@@ -186,10 +186,12 @@ def account(self, redirect=None, **post):
         })
 
         if post and request.httprequest.method == 'POST':
-            if not partner.can_edit_vat():
+            if not partner.can_edit_vat() and partner.country_id:
                 post['country_id'] = str(partner.country_id.id)
 
             error, error_message = self.details_form_validate(post)
+            if error and not partner.country_id:
+                post['country_id'] = 'False'
```

Comment:
Idea was to use `False` instead of `'False'` by doing `str(partner.country_id.id) if partner.country_id else False` otherwise it's the same issue as before. I also don't really see the point in resetting the country if it failed for another reason.

No need to touch the xml condition then, or to add a weird `== 'False'` in the other one.